examples/onnxruntime/pytorch/question-answering/run_qa.py

```diff
@@ -0,0 +1,735 @@
+    trainer = QuestionAnsweringTrainer(
+        model=model,
+        args=training_args,
+        train_dataset=train_dataset if training_args.do_train else None,
+        eval_dataset=eval_dataset if training_args.do_eval else None,
+        eval_examples=eval_examples if training_args.do_eval else None,
+        tokenizer=tokenizer,
+        data_collator=data_collator,
+        post_process_function=post_processing_function,
+        compute_metrics=compute_metrics,
+    )
+
+    ort_config = ORTConfig(
+        opset=optim_args.opset,
+        opt_level=optim_args.opt_level,
+        only_onnxruntime=optim_args.only_onnxruntime,
+        quantization_approach=optim_args.quantization_approach,
+        per_channel=optim_args.per_channel,
+        reduce_range=optim_args.reduce_range,
+        weight_type=optim_args.weight_type,
+        activation_type="uint8",
+        quant_format=optim_args.quant_format,
+        calibration_method=optim_args.calibration_method,
+        optimize_model=optim_args.optimize,
+        max_samples=optim_args.max_calib_samples,
+        calib_batch_size=training_args.per_device_eval_batch_size,
+        seed=training_args.seed,
+    )
+
+    eval_dataloader = trainer.get_eval_dataloader(eval_dataset)
+    output_model = trainer.evaluation_loop(eval_dataloader, description="evaluation")
+    model_eval_preds = post_processing_function(eval_examples, eval_dataset, output_model.predictions)
+    metrics_model = compute_metrics(model_eval_preds)
+    results_model = metrics_model.get(optim_args.metric_name)
+    output_dir = Path(training_args.output_dir)
+
+    if optim_args.quantize:
+        calib_dataset = None
+        if optim_args.quantization_approach == "static":
+            if optim_args.calib_dataset_split not in raw_datasets:
+                raise ValueError(f"The split {optim_args.calib_dataset_split} is not present in the provided dataset.")
+            calib_dataset = raw_datasets[optim_args.calib_dataset_split]
+            if optim_args.max_calib_samples is not None:
+                calib_dataset = calib_dataset.select(range(optim_args.max_calib_samples))
+            calib_dataset = calib_dataset.map(
+                prepare_validation_features,
+                batched=True,
+                num_proc=data_args.preprocessing_num_workers,
+                remove_columns=column_names,
+                load_from_cache_file=not data_args.overwrite_cache,
+                desc="Running tokenizer on calibration dataset",
+            )
+
+        quantizer = ORTQuantizer(ort_config, calib_dataset=calib_dataset, data_collator=data_collator)
+        quantizer.fit(
+            model_args.model_name_or_path,
+            output_dir,
+            feature="question-answering",
+            cache_dir=model_args.cache_dir,
+        )
+        onnx_config = quantizer.onnx_config
+        opt_model_path = output_dir.joinpath("model-quantized.onnx")
+
+    elif optim_args.optimize:
```

Comment:
Is it possible to apply quantization as well as optimization?

Reply:
Optimization can be applied with quantization (when `optimize_model` attribute of `ORTConfig` is set to True), however, `ORTQuantizer` and `ORTOptimizer` are not very compatible (in the sense that if we first use `ORTOptimizer`  and then `ORTQuantizer` , a lot of operators will not be quantized)